core/iwasm/compilation/aot_emit_conversion.c

```diff
@@ -347,17 +347,8 @@ aot_compile_op_i32_trunc_f32(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
 
     POP_F32(value);
 
-    if (!comp_ctx->is_indirect_mode) {
-        if (sign) {
-            min_value = F32_CONST(-2147483904.0f);
-            max_value = F32_CONST(2147483648.0f);
-        }
-        else {
-            min_value = F32_CONST(-1.0f);
-            max_value = F32_CONST(4294967296.0f);
-        }
-    }
-    else {
+    if (comp_ctx->is_indirect_mode
+        && aot_intrinsic_check_capability(comp_ctx, "f32.const")) {
```

Comment:
Perhaps there ought to be a function to eliminate a particular intrinsic capability, such as `remove_intrinsic_capability(comp_ctx, AOT_INTRINSIC_FLAG_F32_CMP)`?